packages/darklang/languageTools/writtenTypes.dark

```diff
@@ -340,6 +337,27 @@ module Darklang =
 
         | EFnName of Range * name: QualifiedFnIdentifier
 
+      type PipeExpr =
+        | EPipeInfix of Range * op: (Range * Infix) * Expr
+        | EPipeLambda of
+          Range *
+          pats: List<LetPattern> *
+          body: Expr *
+          keywordFun: Range *
+          symbolArrow: Range
+        | EPipeEnum of
+          Range *
+          typeName: (Range * List<String>) *
+          caseName: (Range * String) *
+          fields: List<Expr> *
+          /// between the typeName and the caseName
+          symbolDot: Range
+        | EPipeFnCall of
+          Range *
+          fnName: Name *
+          typeArgs: List<TypeReference.TypeReference> *
```

Comment:
Note to self: I remember reading a comment about removing `typeArgs` from this, try to find the comment

Reply:
I recall making that comment somewhere, some time ago. I think I only added the comment to simplify the grammar a bit, until it was complete. Long-term, I think you _should_ be able to `"1" |> Json.parse<Int>`, in which case `typeArgs` here is OK.